### PR TITLE
[17497] Update uncrustify workflow using eProsima action

### PR DESCRIPTION
**Pull request by JLBuenoLopez-eProsima**
This PR should be merged once eProsima/eProsima-CI#4 has been merged so the workflow keeps working.

### CODE REVIEW

File: .github/workflows/uncrustify.yml [hunk 1]
```diff
@@ -7,66 +7,15 @@ on:
     paths:
       - '**.cpp'
       - '**.hpp'
-      - '.h'
-      - '.cxx'
+      - '*.h'
```
Comment (jparisu) [inline .github/workflows/uncrustify.yml:10]: should not be ** ?

**Review by jparisu — CHANGES_REQUESTED** [review]

File: .github/workflows/uncrustify.yml [hunk 2]
```diff
@@ -7,66 +7,15 @@ on:
...
-          cd src/Fast-DDS-QoS-Profiles-Manager
-          ament_uncrustify \
-            -c ../../uncrustify.cfg \
-            --language CPP \
-            --xunit-file ../../uncrustify_results.xml \
-            ${MODIFIED_FILES}
+      - name: eProsima uncrustify
+        uses: eProsima/eProsima-CI/actions/uncrustify@main
 
       - name: Upload uncrustify results
```
Comment (jparisu) [inline .github/workflows/uncrustify.yml:20]: This upload should have at least a `path` argument. Also a name is pretty advisable.

Comment (jparisu) [inline .github/workflows/uncrustify.yml:20]: Use new @v3

**Review by jparisu — CHANGES_REQUESTED** [review]

File: .github/workflows/uncrustify.yml (hunk 2 above)
Comment (JLBuenoLopez-eProsima) [reply .github/workflows/uncrustify.yml:20]: These arguments are already there.

**Review by jparisu — APPROVED** [review]
LGTM

**Pull request merged**